### PR TITLE
Added testcase to capture the ebizzy workload

### PR DESCRIPTION
**Pull request by Krishan-Saraswat**
test_workload test case is added to capture the output of ebizzy workload using perf_top and validating it.
Added a readme file on how to run the workload test case separately.

### CODE REVIEW

File: perf/perf_top.py [hunk 1]
```diff
@@ -56,12 +61,46 @@ def setUp(self):
...
+        self.temp_file = tempfile.NamedTemporaryFile().name
+
         # Getting the parameters from yaml file
         self.option = self.params.get('option', default='')
 
         # Clear the dmesg by that we can capture delta at the end of the test
         dmesg.clear_dmesg()
 
+    def test_workload(self):
+        # Get ebizzy workload and build
```
Comment (Naresh-ibm) [inline perf/perf_top.py:74]: @Krishan-Saraswat please have proper doc-string

File: perf/perf_top.py [hunk 2]
```diff
@@ -56,12 +61,46 @@ def setUp(self):
...
+        thread2.start()
+        # Wait for both threads to finish
+        thread1.join()
+        thread2.join()
+        # Main thread continues here
+        output = subprocess.check_output("grep ebizzy %s" % self.temp_file, shell=True, stderr=subprocess.STDOUT)
+        if not output:
+            self.fail("ebizzy workload not captured in perf top")
+
+    def run_workload(self):
```
Comment (Naresh-ibm) [inline perf/perf_top.py:98]: same here too

File: perf/perf_top.py [hunk 3]
```diff
@@ -56,12 +61,46 @@ def setUp(self):
...
+        process.run("./ebizzy -S1 -s1024 -t1", shell=True)
+
+    def capture_top_output(self):
```
Comment (Naresh-ibm) [inline perf/perf_top.py:101]: same

File: perf/readme [hunk 4]
```diff
@@ -0,0 +1,6 @@
+perf_top has two functions to run:
+test_workload will test perf_top by running ebizzy workload. To run this no need to pass yaml file, can be run like:
+avocado run <test_name:class_name.function_name>
+avocado run perf_top.py:perf_top.test_workload
```
Comment (Naresh-ibm) [inline perf/readme:4]: make this as `example:`

File: perf/readme [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+test_top will test all different top options with the help of yaml file. It can be run like:
+avocado run --max-parallel-tasks=1 perf_top.py:perf_top.test_top -m perf_top.py.data/perf_top.yaml
```
Comment (Naresh-ibm) [inline perf/readme:6]: add the different yaml parameter and how to fetch the different values for those parameters.
check any other Readme file for reference

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

File: perf/perf_top.py [hunk 6]
```diff
@@ -56,12 +61,54 @@ def setUp(self):
...
+        Get ebizzy workload as a tar file and build it.
+        """
+        url = 'https://sourceforge.net/projects/ebizzy/files/ebizzy/0.3/ebizzy-0.3.tar.gz'
+        tarball = self.fetch_asset(url, expire='7d')
+        archive.extract(tarball, self.workdir)
+        version = os.path.basename(tarball.split('.tar.')[0])
+        self.sourcedir = os.path.join(self.workdir, version)
+        os.chdir(self.sourcedir)
+        process.run("./configure")
+        build.make(self.sourcedir, extra_args='LDFLAGS=-static')
```
Comment (abdhaleegit) [inline perf/perf_top.py:84]: As discussed yesterday, we can still add this as a scenario in a existing cfg, making sure ebizzy is modified to run with any other tests as this is a workload

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
Just think about reusing ebizzy test instead of adding here

**Comment by Krishan-Saraswat** [comment]
Creating a fresh pull request for adding workload.

**Pull request closed without merging**